### PR TITLE
chore(metadata): metadata no longer has owner_guid and enabled

### LINKED ISSUE #10636: User delete causes owned metadata/annotation deletes
While it's makes good sense to delete the user's content, and MD/annotations on that content, it's not clear that the default behavior should be to also delete MD/annotations on *other* content.

We [started deleting owned MD/annotations](https://github.com/Elgg/Elgg/commit/8b5fd806d6871aaeb179376328a8af215ed44f88) for 1.6, because comments (which were then annotations) were being left over and causing problems.

Now that we'd like to move 2nd entity attributes to metadata (#10629), this poses some danger. A user can update an entity's attribute (now stored as MD) and if that user is deleted, the entity attribute will be deleted too.

My preferred solution would be to strip metadata specifically down to simply a key-value store, no access control (already a goal) but also no owner. With this change, an entity's attributes would not be in danger of deletion depending on who last touched them.

### PR DESCRIPTION
**Pull request by jdalsem**
- [x] create phinx upgrade to drop enabled and owner_guid

Fixes #10636 -> no longer deleting owned metadata

### CODE REVIEW

File: engine/lib/metadata.php [hunk 1]
```diff
@@ -58,16 +56,16 @@ function elgg_delete_metadata_by_id($id) {
...
- * @param null   $ignored        This argument is not used
+ * @param int    $ignored1       This argument is not used
+ * @param null   $ignored2       This argument is not used
  * @param bool   $allow_multiple Allow multiple values for one key. Default is false
  *
  * @return int|false id of metadata or false if failure
  */
-function create_metadata($entity_guid, $name, $value, $value_type = '', $owner_guid = 0,
-		$ignored = null, $allow_multiple = false) {
+function create_metadata($entity_guid, $name, $value, $value_type = '', $ignored1 = null,
```
Comment (hypeJunction) [inline engine/lib/metadata.php]: This function no longer makes sense. Better deprecate and use entity setter

File: engine/lib/metadata.php [hunk 2]
```diff
@@ -77,13 +75,11 @@ function create_metadata($entity_guid, $name, $value, $value_type = '', $owner_g
...
  * @param string $value      Metadata value
  * @param string $value_type Value type
- * @param int    $owner_guid Owner guid
  *
  * @return bool
  */
-function update_metadata($id, $name, $value, $value_type, $owner_guid) {
-	return _elgg_services()->metadataTable->update($id, $name, $value,
-		$value_type, $owner_guid);
+function update_metadata($id, $name, $value, $value_type) {
```
Comment (hypeJunction) [inline engine/lib/metadata.php]: Same, either use ElggMetadata instance or entity setter

File: engine/lib/metadata.php [hunk 3]
```diff
@@ -96,17 +92,17 @@ function update_metadata($id, $name, $value, $value_type, $owner_guid) {
...
- * @param null   $ignored         This argument is not used
+ * @param int    $ignored1        GUID of entity that owns the metadata
+ * @param null   $ignored2        This argument is not used
  * @param bool   $allow_multiple  Allow multiple values for one key. Default is false
  *
  * @return bool
  */
-function create_metadata_from_array($entity_guid, array $name_and_values, $value_type, $owner_guid,
-		$ignored = null, $allow_multiple = false) {
+function create_metadata_from_array($entity_guid, array $name_and_values, $value_type, $ignored1 = null,
```
Comment (hypeJunction) [inline engine/lib/metadata.php]: Same

File: engine/lib/metadata.php (hunk 1 above)
Comment (jdalsem) [reply engine/lib/metadata.php]: good points... did not think about that... what to do with `$multiple`?

Comment (hypeJunction) [reply engine/lib/metadata.php]: Well multiple is same as setting value to an array, no?

**Comment by hypeJunction** [comment]
We shouldn't use phinx to drop columns, as people may loose the ability to transfer their data to different storage. Let's schedule removal for 4.0

**Comment by hypeJunction** [comment]
Optionally we can create a backup metadata table clone before dropping

**Comment by jdalsem** [comment]
> We shouldn't use phinx to drop columns, as people may loose the ability to transfer their data to different storage. Let's schedule removal for 4.0

than lets change the default value and/or the fact it can be null... otherwise we need sql code to set defaults

**Comment by hypeJunction** [comment]
After you are done with this I will take a stab at finishing the search service. Should be much easier now

**Comment by jdalsem** [comment]
i am also thinking about the ability to use `elgg_get_entities` for metadata, privatesettings and relationship automatically without the need to call the _correct_ function. This can be done in a very basic way, so we can deprecate the secondary functions somewhere during 3.x.

File: engine/lib/metadata.php (hunk 1 above)
Comment (jdalsem) [reply engine/lib/metadata.php]: going to deprecate this in a separate PR

**Pull request merged**